pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/text/LuceneTextIndexReader.java

```diff
@@ -142,18 +152,17 @@ public MutableRoaringBitmap getDocIds(Object value) {
    *
    * TODO: Explore optimizing this path to avoid building the second bitmap
    */
-  private MutableRoaringBitmap getPinotDocIds(MutableRoaringBitmap luceneDocIds) {
+  private MutableRoaringBitmap getPinotDocIdsFromMappingFile(MutableRoaringBitmap luceneDocIds) {
     IntIterator luceneDocIDIterator = luceneDocIds.getIntIterator();
     MutableRoaringBitmap actualDocIDs = new MutableRoaringBitmap();
     try {
       while (luceneDocIDIterator.hasNext()) {
         int luceneDocId = luceneDocIDIterator.next();
-        Document document = _indexSearcher.doc(luceneDocId);
-        int pinotDocId = Integer.valueOf(document.get(LuceneTextIndexCreator.LUCENE_INDEX_DOC_ID_COLUMN_NAME));
+        int pinotDocId = _docIdReaderWriter.getInt(luceneDocId * Integer.BYTES);
         actualDocIDs.add(pinotDocId);
       }
     } catch (Exception e) {
-      throw new RuntimeException("Error: failed while retrieving document from index: " + e);
+      throw new RuntimeException("Error: failed while retrieving pinot doc id from mapping file: " + e);
```

Comment:
No need to catch, you can directly throw the exception

Reply:
done